lib/luwra/usertypes.hpp

```diff
@@ -119,8 +119,9 @@ namespace internal {
  */
 template <typename U>
 struct Value<U&> {
-	using T = internal::StripUserType<U>;
-
+	using X = internal::StripUserType<U>;
+	using T = typename std::conditional<std::is_function<U>::value && !std::is_pointer<U>::value, U*, U>::type;
```

Comment:
I had mistakenly through this specialization could be used, before realizing that arbitrary functions / lambdas couldn't be used here.